### PR TITLE
KF-2356 Use Rock

### PR DESCRIPTION
**Pull request by misohu**
Use rock instead of docker image

### CODE REVIEW

File: metadata.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ resources:
   oci-image:
     type: oci-image
     description: Backing OCI image
-    upstream-source: docker.io/ubuntu/mlflow:2.1.1_1.0-22.04
+    upstream-source: docker.io/charmedkubeflow/mlflow:v2.1.1_22.04_1
```
Comment (i-chvets) [inline metadata.yaml:16]: Was it manually published?

**Comment by i-chvets** [comment]
Since we are integrating manually published ROCK, we need to provide at least some instructions on how it was published. This is until common CI for publishing gets approved (it's been in review for quite some time).

File: metadata.yaml (hunk 1 above)
Comment (misohu) [reply metadata.yaml:16]: yes

**Comment by misohu** [comment]
@i-chvets I have updated the spec with my example as we discussed https://docs.google.com/document/d/1wKd6lKGVnaPdwxSSHWqCZzjeCCB-p1UhSh_Vp6a6gng/edit#heading=h.w9czzrb1b4nw

**Review by DnPlas — APPROVED** [review]
LGTM, thanks @misohu

**Pull request merged**